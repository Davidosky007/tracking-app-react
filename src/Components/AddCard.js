import PropTypes from 'prop-types';

const AddCard = (props) => {
  const {
    id, title, submitHandler, changeHandler, value,
  } = props;

  const handleSubmit = (e, id) => {
    submitHandler(e.target.previousSibling.id, id);
  };

  const handleChange = (e) => {
    changeHandler(e.target.id, e.target.value);
  };

  return (
    <div className="add-card flex space-between align-center m-b-20 p-l-10 p-r-10">
      <div className="color-gray medium input-title flex">{title}</div>
      <input className="input" id={title} type="number" placeholder="0" value={value} onChange={(e) => handleChange(e)} />
      <button className="submit-button color-dark-gray medium" type="submit" onClick={(e) => handleSubmit(e, id)}>+</button>
    </div>
  );
};

AddCard.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
  submitHandler: PropTypes.func.isRequired,
  changeHandler: PropTypes.func.isRequired,
};

AddCard.defaultProps = {
  value: '',
};

export default AddCard;
