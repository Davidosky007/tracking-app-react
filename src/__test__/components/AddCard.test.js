import renderer from 'react-test-renderer';
import AddCard from '../../Components/AddCard';

describe('AddCard', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AddCard
      id={1}
      title="Title"
      value="5"
      submitHandler={() => {}}
      changeHandler={() => {}}
    />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
