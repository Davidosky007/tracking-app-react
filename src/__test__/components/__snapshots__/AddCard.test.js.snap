// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCard renders correctly 1`] = `
<div
  className="add-card flex space-between align-center m-b-20 p-l-10 p-r-10"
>
  <div
    className="color-gray medium input-title flex"
  >
    Title
  </div>
  <input
    className="input"
    id="Title"
    onChange={[Function]}
    placeholder="0"
    type="number"
    value="5"
  />
  <button
    className="submit-button color-dark-gray medium"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
</div>
`;
